fhir-path/src/main/java/com/ibm/fhir/path/function/ConformsToFunction.java

```diff
@@ -43,20 +55,36 @@ public int getMaxArity() {
             throw new IllegalArgumentException("The argument to the 'conformsTo' function must be a string");
         }
         
-        Class<?> modelClass = getSingleton(context).type().modelClass();
+        FHIRPathNode node = getSingleton(context);
         
+        if (node.isResourceNode() && node.asResourceNode().resource() == null) {
+            return SINGLETON_TRUE;
+        }
+        
+        Class<?> modelClass = node.type().modelClass();
         String url = getStringValue(arguments.get(0)).string();
-        if (url.startsWith(HL7_STRUCTURE_DEFINITION_URL_PREFIX)) {
-            String s = url.substring(HL7_STRUCTURE_DEFINITION_URL_PREFIX.length());
-            if (modelClass != null && s.equals(modelClass.getSimpleName())) {
-                return SINGLETON_TRUE;
+                
+        if (modelClass != null && FHIRRegistry.getInstance().hasResource(url)) {
+            if (url.startsWith(HL7_STRUCTURE_DEFINITION_URL_PREFIX)) {
+                String s = url.substring(HL7_STRUCTURE_DEFINITION_URL_PREFIX.length());
+                if (s.equals(modelClass.getSimpleName())) {
+                    return SINGLETON_TRUE;
+                }
+            }
+            
+            FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();    
+            for (Constraint constraint : ProfileSupport.getConstraints(url, modelClass)) {
+                try {
+                    Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, constraint.expression(), context);
+                    if (evaluatesToBoolean(result) && isFalse(result)) {
+                        return SINGLETON_FALSE;
+                    }
+                } catch (FHIRPathException e) {
+                    // TODO: log this
```

Comment:
we have https://github.com/IBM/FHIR/issues/187 for cleaning up TODOs, but obviously better just to add now it if its quick/easy

Reply:
added logging to the catch block